packages/opentelemetry-core/src/utils/environment.ts

```diff
@@ -16,37 +16,109 @@
 
 import { LogLevel } from '../common/types';
 
-export type ENVIRONMENT_MAP = { [key: string]: string | number };
+const DEFAULT_LIST_SEPARATOR = ',';
 
 /**
  * Environment interface to define all names
  */
-export interface ENVIRONMENT {
-  OTEL_LOG_LEVEL?: LogLevel;
-  OTEL_NO_PATCH_MODULES?: string;
+
+type ENVIRONMENT_NUMBERS = {
   OTEL_SAMPLING_PROBABILITY?: number;
   OTEL_SPAN_ATTRIBUTE_COUNT_LIMIT?: number;
   OTEL_SPAN_EVENT_COUNT_LIMIT?: number;
   OTEL_SPAN_LINK_COUNT_LIMIT?: number;
-}
+};
 
-const ENVIRONMENT_NUMBERS: Partial<keyof ENVIRONMENT>[] = [
+const ENVIRONMENT_NUMBERS_KEYS: Partial<keyof ENVIRONMENT_NUMBERS>[] = [
   'OTEL_SAMPLING_PROBABILITY',
   'OTEL_SPAN_ATTRIBUTE_COUNT_LIMIT',
   'OTEL_SPAN_EVENT_COUNT_LIMIT',
   'OTEL_SPAN_LINK_COUNT_LIMIT',
 ];
 
+function isEnvVarANumber(key: unknown): key is keyof ENVIRONMENT_NUMBERS {
+  return ENVIRONMENT_NUMBERS_KEYS.includes(key as keyof ENVIRONMENT_NUMBERS);
+}
+
+type ENVIRONMENT_LISTS = {
+  OTEL_NO_PATCH_MODULES?: string[];
+};
+
+const ENVIRONMENT_LISTS_KEYS: Partial<keyof ENVIRONMENT_LISTS>[] = [
+  'OTEL_NO_PATCH_MODULES',
+];
+
+function isEnvVarAList(key: unknown): key is keyof ENVIRONMENT_LISTS {
+  return ENVIRONMENT_LISTS_KEYS.includes(key as keyof ENVIRONMENT_LISTS);
+}
+
+type ENVIRONMENT_STRINGLIKES = {
+  OTEL_LOG_LEVEL?: LogLevel;
+  JAEGER_ENDPOINT?: string;
```

Comment:
should we support both (by rewriting) or break compatibility and only support the official spec versions?

Reply:
also, this issue is about moving all accesses to `process.env.` to `getEnv()`, not just the implementation of spec env vars

Reply:
before GA I think it is fine to support the official specs only, otherwise we will have problems like what if both property are different which one should win etc., So that's why before GA we should focus on being spec compliant as much as possible even if that means we break "backward compatibility" although I don't see here any breaking - you will have to only provide a different name in environment.